src/bluesky/protocols.py

```diff
@@ -521,13 +531,64 @@ def hints(self) -> Hints:
 
 
 @runtime_checkable
-class NamedMovable(Movable[T_co], HasHints, Protocol):
+class NamedChild(HasParent, HasName, Protocol):
+    """A child object that has a name and a parent device."""
+
+
+# This is a convenience alias for when you need a Movable that also has a name.
+# This is required in plans.list_scan.
+@runtime_checkable
+class NamedMovable(Movable[T_co], HasName, Protocol):
+    """A movable object that has a name."""
+
+
+@runtime_checkable
+class NamedChildMovable(NamedChild, Movable[T_co], Protocol):
+    """A movable object that has a name and a parent device."""
+
+
+@runtime_checkable
+class NamedChildMovableAndStageable(NamedChild, Movable[T_co], Stageable, Protocol):
+    """A movable and stageable object that has a name and a parent device."""
+
+
+@runtime_checkable
+class HintedMovable(Movable[T_co], HasHints, Protocol):
     """A movable object that has a name and hints."""
 
-    ...
 
+# TODO if this is too ugly, just include HasParent into all devices
+# Adding this here to make it easy to search the codebase for places where it is needed.
```

Comment:
Hmm, I wonder if we should include `HasParent` and `HasName` into all the protocols? Would that remove all the ugliness here?